src/components/HeartRateZoneGraph.vue

```diff
@@ -354,21 +359,27 @@ export default {
       let totalSeconds = 0
 
       // Process each data point and calculate heart rate zone and total time
+      // TODO optimize calculation
       sessions.forEach((session) => {
-        if (!session.records) return
+        if (session.records == null) return
+        console.time('totalSteps')
         for (let i = 0; i < session.records.length - 1; i++) {
           const entry = session.records[i]
-          const nextEntry = session.records[i + 1]
+          if (entry == null || entry.heartRate == null) continue
```

Comment:
actually entry will never be null, it's guaranteed. but this is okay too, no worries.

Reply:
> actually entry will never be null, it's guaranteed. but this is okay too, no worries.

i think this leftover from previous one that Record can be null (cause linter red), i'll check more on this